Mage.Sets/src/mage/cards/v/VrestinMenoptraLeader.java

```diff
@@ -0,0 +1,74 @@
+package mage.cards.v;
+
+import mage.MageInt;
+import mage.abilities.common.AttacksWithCreaturesTriggeredAbility;
+import mage.abilities.common.EntersBattlefieldAbility;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.dynamicvalue.common.GetXValue;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.effects.common.EntersBattlefieldWithXCountersEffect;
+import mage.abilities.effects.common.counter.AddCountersAllEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.counters.CounterType;
+import mage.filter.common.FilterAttackingCreature;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.permanent.token.AlienInsectToken;
+
+import java.util.UUID;
+
+/**
+ * @author Sidorovich77
+ */
+public final class VrestinMenoptraLeader extends CardImpl {
+
+    private static final FilterAttackingCreature filter = new FilterAttackingCreature();
+
+    static {
+        filter.add(SubType.INSECT.getPredicate());
+    }
+
+    public VrestinMenoptraLeader(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{X}{G}{G}{W}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.ALIEN);
+        this.subtype.add(SubType.INSECT);
+        this.subtype.add(SubType.SCOUT);
+        this.power = new MageInt(0);
+        this.toughness = new MageInt(0);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Vrestin enters with X +1/+1 counters on it.
+        this.addAbility(new EntersBattlefieldAbility(
+                new EntersBattlefieldWithXCountersEffect(CounterType.P1P1.createInstance())
+        ));
+
+        // When Vrestin enters, create X 1/1 green and white Alien Insect creature tokens with flying.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(
+                new CreateTokenEffect(new AlienInsectToken(), GetXValue.instance)
+                        .setText("create X 1/1 green and white Alien Insect creature tokens with flying.")
+        ));
+
+        // Whenever you attack with one or more Insects, put a +1/+1 counter on each of them.
+        this.addAbility(new AttacksWithCreaturesTriggeredAbility((new AddCountersAllEffect(CounterType.P1P1.createInstance(),
+                filter).setText("put a +1/+1 counter on each of them.")), 1,
+                new FilterCreaturePermanent(SubType.INSECT, "Insects")).setTriggerPhrase("Whenever you attack with one or more Insects, "));
```

Comment:
this isn't right - you shouldn't have a separate filter. Use the setTargetPointer argument in the constructor of the triggered ability, and AddCountersTargetEffect. And should not need to set trigger phrase (set text in the static filter)